shimmingtoolbox/load_nifti.py

```diff
@@ -151,30 +152,89 @@ def load_nifti(path_data, modality='phase'):
     return niftis, info, json_info
 
 
-def read_nii(nii_path, auto_scale=True):
+def read_nii(fname_nifti, auto_scale=True):
     """ Reads a nifti file and returns the corresponding image and info. Also returns the associated json data.
     Args:
-        nii_path (str): direct path to the .nii or .nii.gz file that is going to be read
+        fname_nifti (str): direct path to the .nii or .nii.gz file that is going to be read
         auto_scale (:obj:`bool`, optional): Tells if scaling is done before return
     Returns:
         info (Nifti1Image): Objet containing various data about the nifti file (returned by nibabel.load)
         json_data (dict): Contains the different fields present in the json file corresponding to the nifti file
-        image (ndarray): Image contained in the read nifti file. Siemens phase images are rescaled between 0 and 2pi.
+        image (ndarray): For B0-maps, image contained in the nifti. Siemens phase images are rescaled between 0 and 2pi.
+                         For RF-maps, complex array of dimension (x, y, slice, coil) with phase between -pi and pi.
     """
 
-    info = nib.load(nii_path)
-
+    info = nib.load(fname_nifti)
     # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
-    json_path = nii_path.split('.nii')[0] + '.json'
+    json_path = fname_nifti.split('.nii')[0] + '.json'
 
     if os.path.isfile(json_path):
         json_data = json.load(open(json_path))
     else:
-        raise ValueError('Missing json file')
+        raise ValueError("Missing json file")
 
+    # Store nifti image in a numpy array
     image = np.asarray(info.dataobj)
+
     if auto_scale:
-        if ('Manufacturer' in json_data) and (json_data['Manufacturer'] == 'Siemens')\
+
+        # If Siemens' TurboFLASH B1 mapping (dcm2niix cannot separate phase and magnitude for this sequence)
+        if ('SequenceName' in json_data) and 'tfl2d1_16' in json_data['SequenceName']:
```

Comment:
and if not?

Reply:
Once again no scaling will be done. But it could be interesting to add a warning saying to the user that no scaling was performed because the type of nifti is unknown

Reply:
or an error...

Reply:
@po09i any pros/cons for choosing to throw an error instead of just a log if the nifti type is not recognized ?

Reply:
Since `auto_scale` is True by default, I think it makes sense to output a warning so that the function can still be usable by default if the rescaling is not implemented.